### PR TITLE
Refactor main main.go file

### PR DESCRIPTION
**Pull request by benzend**
moved most of the logic into the routes dir.

another big thing is that there's a Router utility that allows us to set the Post / Get / Put / Delete methods with ease, similar to Express.

### CODE REVIEW

**Review by benzend — COMMENTED** [review]
A couple of things:

- Created a new `router` util that allows us to have a closer syntax to express and manage Get, Post, Put, Delete methods with ease. This also segues us into being able to inject middlewares.
- Added a /heathcheck route to confirm server health
- Removed the code that was breaking the application - there were missing files and imports
- Last but not least, the main goal for this PR - to clean up the main.go file.

File: backend/ctx/key.go
Comment (ghost) [inline backend/ctx/key.go:1]: this is bloat

Comment (benzend) [reply backend/ctx/key.go:1]: ur bloat

Comment (benzend) [reply backend/ctx/key.go:1]: oh shit that's actually a good point

Comment (benzend) [reply backend/ctx/key.go:1]: removed

**Pull request merged**